connect-wallet-nextjs/pages/ShowNFTs.js

```diff
@@ -0,0 +1,41 @@
+import styles from '../styles/Home.module.css'
+import { useMetaplex } from "./useMetaplex";
+import { useState } from "react";
+
+export const ShowNFTs = () => {
+    const { metaplex } = useMetaplex();
+
+    const [nft, setNft] = useState(null);
+    const onClick = async () => {
+        const myNfts = await metaplex.nfts().findAllByOwner(metaplex.identity().publicKey);
+        let randIdx = Math.floor(Math.random() * myNfts.length);
+        setNft(myNfts[randIdx]);
```

Comment:
The endpoint `findAllByOwner` returns an array of `Nft`s whose metadata is not loaded yet. You need to load the metadata before calling `setNft` otherwise, the image will never be present. Please try this example with a wallet that has NFTs with images.